pkg/storage/watch/watch.go

```diff
@@ -0,0 +1,196 @@
+package watch
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/rjeczalik/notify"
```

Comment:
any specific reason you chose this package?
it seems like it doesn't support recursive watching, right?

Reply:
`inotify` itself doesn't support recursive watches. For this reason there are no good golang packages supporting that feature.

The few small ones that might support it don't provide low-level enough access to `inotify` to monitor `IN_CLOSE_WRITE`, which is the ideal way to fire a single event after the whole write is complete instead of a flood of events for every small write operation. https://github.com/rjeczalik/notify doesn't support recursive watching, but provides the `IN_CLOSE_WRITE` I need.

`watcher` _does_ however support recursive watching, it will dynamically add and remove watches for all subdirectories:
```golang
if fi.IsDir() {
	if err := w.addWatch(event.Path()); err != nil {
```